### PR TITLE
Blob Create Snapshot Returns a Client

### LINKED ISSUE #4453: createSnapshot Returning Client
createSnapshot should return a client to interact with the snapshot as a convenience.

### PR DESCRIPTION
**Pull request by alzimmermsft**
Fixes #4453 

createSnapshot in Storage Blobs returns a new BlobClient that is used to interact with the snapshot that was created. Convenience method was added to create a new client by passing a snapshot instead of needing to create a new builder and configuring it.

### CODE REVIEW

File: sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobAsyncClient.java [hunk 1]
```diff
@@ -703,17 +721,18 @@ private void downloadToFileCleanup(AsynchronousFileChannel channel) {
...
+    public Mono<Response<BlobAsyncClient>> createSnapshot(Metadata metadata, BlobAccessConditions accessConditions) {
         metadata = metadata == null ? new Metadata() : metadata;
         accessConditions = accessConditions == null ? new BlobAccessConditions() : accessConditions;
 
         return postProcessResponse(this.azureBlobStorage.blobs().createSnapshotWithRestResponseAsync(
             null, null, null, metadata, null, null,
             null, null, accessConditions.modifiedAccessConditions(),
             accessConditions.leaseAccessConditions(), Context.NONE))
-            .map(rb -> new SimpleResponse<>(rb, rb.deserializedHeaders().snapshot()));
+            .map(rb -> new SimpleResponse<>(rb, this.getSnapshotClient(rb.deserializedHeaders().snapshot())));
```
Comment (sima-zhu) [inline sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobAsyncClient.java]: Not very important but a thought, can we have getSnapshotId API?

File: sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/SASTest.groovy [hunk 2]
```diff
@@ -135,12 +135,12 @@ class SASTest extends APISpec {
         when:
         def sas = bu.generateSAS(null, permissions, expiryTime, startTime, null, sasProtocol, ipRange, cacheControl, contentDisposition, contentEncoding, contentLanguage, contentType)
 
-        def client = new BlobClientBuilder()
-            .endpoint(cu.getContainerUrl().toString())
+        def builder = new BlobClientBuilder()
```
Comment (sima-zhu) [inline sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/SASTest.groovy]: the builder did not use below. You can def and build a client directly.

File: sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/SASTest.groovy [hunk 3]
```diff
@@ -187,13 +187,13 @@ class SASTest extends APISpec {
         when:
         def sas = snapshotBlob.generateSAS(null, permissions, expiryTime, startTime, null, sasProtocol, ipRange, cacheControl, contentDisposition, contentEncoding, contentLanguage, contentType)
 
-        def client = new BlobClientBuilder()
-            .endpoint(cu.getContainerUrl().toString())
+        def builder = new BlobClientBuilder()
```
Comment (sima-zhu) [inline sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/SASTest.groovy]: same here.

File: sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobAsyncClient.java (hunk 1 above)
Comment (alzimmermsft) [reply sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobAsyncClient.java]: There is a getSnapshotId method in the client

**Pull request merged**